Bottom: 635dbb69222e9c2f25c1e18a66a37862a9bdfddc
Top:    9608c3d0519913ba2fcc5a7b95aeeb3e09db4502
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-12-13 02:13:56 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index 40de87f..893ab8e 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2623,7 +2623,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
